aten/src/ATen/FunctionalStorageImpl.h

```diff
@@ -65,8 +65,8 @@ struct ViewMeta {
 class Alias {
   public:
     struct Update {
-        const at::Tensor& new_val;
-        const std::vector<ViewMeta>& view_metas;
+        const at::Tensor new_val;
```

Comment:
To summarize, the problem would happen when one tries to functionalize the following?
```
x = torch.randn(2)
def f(x):
  y = x.view(1, 2)
  y.add_(1)
f(x)
```
x's storage gets an Alias object, `y.add_(1)` pushes an Update object to the Alias that refers to y's `vector<ViewMeta>`, but if `y` goes out of scope then the Alias' update's reference is now invalid.

Reply:
Yep- this snippet of code on its own won't directly hit the issue, but if you try to sync `x` (with `torch._sync(x)`) after the call to `f`, you would try to use the `Update` object which contains the invalid reference that you pointed out.

As for testing - I tried and failed to add a test for it in `test/test_functionalization.py` :(. Those tests in theory should have the exact same problem, because they run a `sync_()` call after calling each function, when the intermediate views go out of scope. But for some reason they don't fail (I'm tentatively attributing this to UB?).